### PR TITLE
Replace tailscale binary with tailscaled unix socket

### PR DESCRIPTION
**Pull request by marwan-at-work**
We currently check for existence of the `tailscale` binary just for guessing at whether you have tailscale installed.  However, we already show a similar message when tsrelay returns a NOT_RUNNING state. 

Furthermore, showing those pop ups on first install when we already have a walkthrough to show you how to install Tailscale is not a pleasant experience.

Therefore, this PR removes the manual checks for the binary and switches the VSCode configuration to be for setting the unix socket

Updates https://github.com/tailscale-dev/vscode-tailscale/issues/76

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,14 +81,12 @@ If the extension isn't working, we recommend following these steps to troublesho
...
-4. Ensure `tailscale` is available in the environment path. You can check this by running `tailscale status` in your CLI; if no command is found, you may need to add the Tailscale executable to your path. Alternatively, you can set its path via the `tailscale.path` setting in VS Code.
+4. If you are running `tailscaled` in a non-default path, you can set its path via the `tailscale.path` setting in VS Code.
 
 ## Configuration
 
-- `tailscale.path`: A path to the `tailscale` executable. If unset, the extension will use
-  the environment path to resolve the `tailscale` executable. If set, the extension
-  will use the supplied path. The path should include the executable name (e.g.
-  `/usr/bin/tailscale`, `C:\Program Files\tailscale\tailscale.exe`).
+- `tailscale.path`: A path to the `tailscaled` unix socket. If unset, the extension will use
```
Comment (tylersmalley) [inline README.md:88]: Think we should change the config to `socketPath`?

**Review by tylersmalley — APPROVED** [review]
Couple small things, but aside from those this LGTM.

Tested the case where Tailscale is not installed. An error is presented on the Funnel panel with a link to install.

image

File: README.md (hunk 1 above)
Comment (marwan-at-work) [reply README.md:88]: @tylersmalley sounds good

File: src/extension.ts [hunk 2]
```diff
@@ -1,55 +1,23 @@
...
-      });
-  }
-
   tailscaleInstance = await Tailscale.withInit(vscode);
 
   // walkthrough completion
   tailscaleInstance.serveStatus().then((status) => {
     // assume if we have any BackendState we are installed
-    vscode.commands.executeCommand('setContext', 'tailscale.walkthroughs.installed', !!commandPath);
+    const isInstalled = status.BackendState !== '';
```
[the remark this answers is not in the document]
Comment (tylersmalley) [reply src/extension.ts:19]: Ah, didn't realize the empty string. 👍

**Pull request merged**